### PR TITLE
Update struct handling section of Jit Opt Doc

### PR DESCRIPTION
**Pull request by CarolEidt**

### CODE REVIEW

**Comment by CarolEidt** [comment]
@dotnet/jit-contrib PTAL - I've included related issues in the first-class-structs doc, and have updated the text here.

Also - can someone remind me how to tell the CI system not to test this?

File: Documentation/performance/JitOptimizerTodoAssessment.md [hunk 1]
```diff
@@ -12,22 +12,42 @@ Big-Ticket Items
...
+   - Eliminate additional cases where structs are unnecessarily forced to memory when passed to or
+     returned from calls, even though there is a one-to-one match between fields and registers.
+     - This impacts the binarytrees benchmark.
+   - Enable non-memory assignments of same-sized structs with different field composition.
+     This is effectively what is required to allow passing of structs in registers (on targets that
+     support it) when the fields don't match the registers.
+
+ - Test and reliability improvements:
+   - Struct promotion stress mode (test mode to improve robustness/reliability) - i.e. eliminate
+     or greatly increase current limitations on the size or number of fields.
```
Comment (mikedn) [inline Documentation/performance/JitOptimizerTodoAssessment.md]: "increase current limitations" sounds strange, I guess it's "increase current limits"?

**Comment by mikedn** [comment]
@CarolEidt Could you take a look at the last paragraph in https://github.com/dotnet/coreclr/pull/15348#issuecomment-348685818 ? That seems like a problem related to passing struct arguments and I'm not sure if it was included and where it fits.

**Comment by CarolEidt** [comment]
@mikedn - I'm out of the office for the next couple of days, but I'll have a look at this when I get back. Some local assignments are already represented as simple as simple assignments, without an indirection, and as we modify struct passing we'll see even fewer of these, so it would seem that the code you're proposing to eliminate wouldn't ever really be useful.

File: Documentation/performance/JitOptimizerTodoAssessment.md [hunk 2]
```diff
@@ -12,22 +12,42 @@ Big-Ticket Items
...
- - Improve handling of System V struct passing (I think we currently insert
-   some unnecessary round-trips through memory at call boundaries due to
-   internal representation issues)
- - Implicit byref parameter promotion w/o shadow copy
-
-We don't have specific benchmarks that we know would jump in response to any of
-these.  May well be able to find some with some looking, though this may be an
-area where current performance-sensitive code avoids structs.
+Most of the work required to improve struct handling in RyuJIT is captered in the [first-class structs](https://github.com/dotnet/coreclr/blob/master/Documentation/design-docs/first-class-structs.md)
+document, though that document needs to be updated to reflect that work that has already been done.
```
Comment (erozenfeld) [inline Documentation/performance/JitOptimizerTodoAssessment.md]: Typo: "that work" --> "the work"

**Pull request merged**